### DISCUSSION TITLE
Password Reset Flow Issue: Unable to Update Password After Redirect

### DISCUSSION THREAD

**Discussion by mbkaj** [supabase/supabase#29305] [Questions]
Hey there

I'm experiencing an issue with the password reset flow in my React application using Supabase authentication. After a user clicks the reset password link in their email, they are correctly redirected to my application with an access token, but when they try to set a new password, the process gets stuck.

## Current Implementation

1. Reset password redirect:
   After clicking the link in the email, users are redirected to:
   ```
   http://localhost:3000/reset-password#access_token=eyJhbGciOiJIUzI1NiIsImtpZCI6ImhrdjhjV0d2MnpFOTZsb0QiLCJ0eXAiOiJKV1QifQ...&expires_at=1726353971&expires_in=3600&refresh_token=3mfJMKmAwJImrA7O_TmrCQ&token_type=bearer&type=recovery
   ```

2. Relevant part of ResetPassword component:
   ```jsx
   useEffect(() => {
     const hashParams = new URLSearchParams(location.hash.slice(1));
     const urlToken = hashParams.get('access_token');
     const type = hashParams.get('type');
     console.log('URL params:', { urlToken, type });
     if (type === 'recovery' && urlToken) {
       setIsResetMode(false);
       setToken(urlToken);
       verifyToken(urlToken);
     }
   }, [location]);

   const handlePasswordChange = async (e) => {
     e.preventDefault();
     setError('');
     setMessage('');
     setIsLoading(true);

     if (password !== confirmPassword) {
       setError('Passwords do not match.');
       setIsLoading(false);
       return;
     }

     try {
       console.log('Attempting to update password...');
       const { data, error } = await supabase.auth.updateUser({ password: password });
       console.log('Update password result:', { data, error });
       
       if (error) throw error;

       console.log('Password updated successfully');
       setMessage('Votre mot de passe a été réinitialisé avec succès.');
       setTimeout(() => {
         console.log('Navigating to signin page...');
         navigate('/signin', { state: { message: 'Password reset successful. Please sign in with your new password.' } });
       }, 3000);
     } catch (error) {
       console.error('Caught error:', error);
       if (error.message.includes('token')) {
         setTokenExpired(true);
         setError('The password reset link is invalid or has expired. Please request a new one.');
       } else {
         setError(error.message || 'An error occurred while resetting your password.');
       }
     } finally {
       console.log('Password change process completed');
       setIsLoading(false);
     }
   };
   ```

## The Issue

The redirect with the access token works correctly, and the component enters the password change mode. However, when the user enters a new password and submits the form, the process gets stuck. The "Change Password" button changes to "Changing..." but never completes the process.

## Expected Behavior

1. User clicks the reset password link in their email.
2. They are redirected to the application's reset password page with an access token.
3. The component recognizes the token and displays the password change form.
4. User enters a new password and confirms it.
5. Upon submission, the password is successfully updated, and the user is redirected to the sign-in page.

## Actual Behavior

1. Steps 1-4 work as expected.
2. Upon submission, the button text changes to "Changing..." but the process never completes.
3. No error message is displayed, and the user is not redirected.

## Questions

1. Is there an issue with how I'm using `supabase.auth.updateUser()` in the `handlePasswordChange` function?
2. Do I need to include the access token when calling `updateUser()`?
3. Are there any known issues with the password reset flow that might cause this behavior?
4. Is there a way to get more detailed error information from the `updateUser()` call?

## Environment

- React version: react@18.3.1
- Supabase JS client version: @supabase/supabase-js@2.45.4
- Browser: all

Any guidance or suggestions would be greatly appreciated.

**Comment by silentworks** [comment]
I have a react example app here with this reset password flow in it https://github.com/silentworks/supabase-by-example/tree/main/react, you don't need to get the token from the URL as the `@supabase/supabase-js` library does this for you when setup correctly. You need to ensure you have your `onAuthStateChange` setup correctly to handle all this. You can check out the example I posted above.
